File: Lead411/Scripts/App/Services/userService.js
routerApp.factory('userService', ['$http', function ($http) {
   
   return {

       // For getting Users List with pagination, searching, sorting
       GetUserList: function (pagingInfo) {
           return $http.get('api/AdminApi/GetUserList?pageNumber=' + pagingInfo.pageNumber + '&PageSize=' + pagingInfo.pageSize + '&sortBy=' + pagingInfo.sortBy + '&reverse=' + pagingInfo.reverse + '&search=' + pagingInfo.search);
        },

        GetUserDetails : function (userMembershipId) {
            return $http.get('api/AdminApi/GetUserDetails?userMembershipId=' + userMembershipId);
        }

    };

}]);

File: ServiceLayer/Services/AdminPanel/UserService.cs
using CoreEntities.CustomModels.AdminPanel;
using ServiceLayer.Interfaces.AdminPanel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceLayer.Services.AdminPanel
{
    public class UserService : IUserService
    {
        private readonly RepositoryLayer.Repositories.Interfaces.IUserRepository _iUserRepository;
        public UserService(RepositoryLayer.Repositories.Interfaces.IUserRepository iUserRepository)
        {
            _iUserRepository = iUserRepository;
        }

        public RegisteredUser GetUserList(int pageNumber, int pageSize, string sortBy, bool reverse, string search)
        {
            return _iUserRepository.GetUserList(pageNumber, pageSize, sortBy, reverse, search);
        }

        public RegisteredUser GetUserDetails(long userMembershipId)
        {
            return _iUserRepository.GetUserDetails(userMembershipId);
        }
    }
}
